external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -138,10 +139,15 @@ class KafkaTestUtils extends Logging {
 
     if (server != null) {
       server.shutdown()
+      server.awaitShutdown()
       server = null
     }
 
-    brokerConf.logDirs.foreach { f => Utils.deleteRecursively(new File(f)) }
+    // On Windows, `logDirs` is left open even after Kafka server above is completely shut-downed
+    // in some cases. It leads to test failures on Windows if these are not ignored.
+    brokerConf.logDirs.map(new File(_))
+      .filterNot(FileUtils.deleteQuietly)
```

Comment:
It seems a Windows specific problem. It holds an exclusive lock. If anything does not close the file, then, it can't be deleted or renamed unlike Linux or Mac. I assume it will be able to be deleted if we call `System.gc()` though.